### PR TITLE
NeutronV2: add protocols

### PR DESCRIPTION
**Pull request by takaishi**
Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

For #293

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://github.com/openstack/neutron-lib/blob/stable/mitaka/neutron_lib/constants.py#L115-L135

### CODE REVIEW

**Comment by takaishi** [comment]
@jrperritt Hello, this is ready for review.

**Comment by splashx** [comment]
@jrperritt up ;$

**Comment by takaishi** [comment]
@jrperritt ping for review

File: openstack/networking/v2/extensions/security/rules/requests.go [hunk 1]
```diff
@@ -47,13 +47,29 @@ type RuleEtherType string
...
+	DirEgress         RuleDirection = "egress"
+	ProtocolTCP       RuleProtocol  = "tcp"
+	ProtocolUDP       RuleProtocol  = "udp"
+	ProtocolICMP      RuleProtocol  = "icmp"
+	ProtocolAH        RuleProtocol  = "ah"
+	ProtocolDCCP      RuleProtocol  = "dccp"
+	ProtocolEGP       RuleProtocol  = "egp"
+	ProtocolGRE       RuleProtocol  = "gre"
+	ProtocolIGMP      RuleProtocol  = "igmp"
+	ProtocolIPv6      RuleProtocol  = "ipv6-encap"
```
Comment (jtopjian) [inline openstack/networking/v2/extensions/security/rules/requests.go]: @takaishi Should this be `ProtocolIPv6Encap`?

Comment (takaishi) [inline openstack/networking/v2/extensions/security/rules/requests.go]: I agree to equalize protocol name gophercloud and neutron.

**Comment by jtopjian** [comment]
@takaishi Thanks for this. There are two more protocols to add. I'll follow-up with a PR to add them.

**Pull request merged**